smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpQueryItemProvider.kt

```diff
@@ -148,27 +156,27 @@ class HttpQueryItemProvider(
                 if (memberTarget is CollectionShape) {
                     renderListOrSet(memberTarget, bindingIndex, memberName, paramName)
                 } else {
-                    renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName)
+                    renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName, true)
                 }
             } else {
-                writer.openBlock("if let $memberName = $memberName {", "}") {
+                writer.openBlock("if let $memberName = value.$memberName {", "}") {
                     if (memberTarget is CollectionShape) {
                         renderListOrSet(memberTarget, bindingIndex, memberName, paramName)
                     } else {
-                        renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName)
+                        renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName, true)
                     }
                 }
             }
         } else {
             if (memberTarget is CollectionShape) {
                 renderListOrSet(memberTarget, bindingIndex, memberName, paramName)
             } else {
-                renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName)
+                renderQueryItem(queryBinding.member, bindingIndex, memberName, paramName, false)
             }
         }
     }
 
-    private fun renderQueryItem(member: MemberShape, bindingIndex: HttpBindingIndex, originalMemberName: String, paramName: String) {
+    private fun renderQueryItem(member: MemberShape, bindingIndex: HttpBindingIndex, originalMemberName: String, paramName: String, unwrapped: Boolean) {
```

Comment:
This function now has an `unwrapped` parameter since it is sometimes called to operate on unwrapped values (i.e. `if let x = value.x`) or wrapped (where `value.x` is accessed directly.)